### DISCUSSION TITLE
Migrating OpenSSL to 1.1.1

### DISCUSSION THREAD

**Discussion by DOpenssl** [openssl/openssl#23787] [Q&A]
I am migrating OpenSSL to 1.1.1, I am getting error for below macros:

M_ASN1_D2I_vars(a, X509_CRL *, X509_CRL_new)
M_ASN1_D2I_Init()
M_ASN1_D2I_start_sequence()
M_ASN1_D2I_get(ret->crl, d2i_X509_CRL_INFO)
M_ASN1_D2I_get(ret->sig_alg, d2i_X509_ALGOR)
M_ASN1_D2I_get(ret->signature, d2i_ASN1_BIT_STRING)
M_ASN1_D2I_Finish(a, X509_CRL_free, ASN1_F_X509_CRL_ADD0_REVOKED);

Can anyone help me to identify alternative macros/functions in openssl1.1.1 for this one?

**Comment by mattcaswell** [comment]
This looks like its just reading the DER for an X509_CRL object. The equivalent would be calling `d2i_X509_CRL`:

https://www.openssl.org/docs/man3.2/man3/d2i_X509_CRL.html

**Reply by DOpenssl** [reply]
Thanks for your answer...

Do you have any example for this one? OR 
Any code reference where we are reading the DER for an X509_CRL object?

**Reply by mattcaswell** [reply]
There's an example of this function being used to read DER data in the test suite:

https://github.com/openssl/openssl/blob/53a8728686663f4fe044cd1a5757f6fcfd777317/test/x509_test.c#L103-L115
